### ISSUE TITLE
Update dynamically on post types that use the Block Editor

### ISSUE THREAD

**Issue by mogmarsh** [alleyinteractive/meta-inspector#28]
### Description

On post types that use the Block Editor (Gutenberg), it would be useful if the inspector reflected the [Post Editor's data](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-editor/).

### Use Case

Currently in the Block Editor, you must Save the post, then refresh to see the latest meta data.

# Acceptance Criteria

- When displayed in Gutenberg, show using a Gutenberg PluginSidebar instead of a classic metabox.
- Create the PluginSidebar to display all meta.
- Each meta field should be in a collapsible panel so it isn't overwhelming.
- Add a custom REST endpoint to query all post meta, even the meta values that aren't exposed to the Block Editor. This MUST be locked down to only logged in users who have the ability to edit the post for which the data is requested (since they would otherwise be able to see this using the classic editor metabox).
- The sidebar should use [usePostMeta](https://github.com/alleyinteractive/alley-scripts/tree/main/packages/block-editor-tools/src/hooks/use-post-meta) to get a value that has been updated by the Block Editor, and fail back to the value from the REST endpoint. Further, a visual indicator should display whether a value has been modified in the current edit session, indicating what is different between the editor view and what is saved to the server.
- Add a button to view meta values in a modal in addition to the sidebar, so that large meta values can be viewed more easily.
- Add JSON Formatter - if a value is JSON, or expressed as JSON because it's a serialized array/object in PHP, pretty-print the JSON to make it easier to read.

Note to implementer: If desired, split out the AC into smaller tickets.

**Comment by mogmarsh** [comment]
I've been discussing with @kevinfodness, and here's what we think this would look like:

- add a PluginSidebar to display all meta. Each meta field can be in a collapsible panel so it isn't overwhelming. If the Block Editor is active, suppress the existing metabox.
- add a custom REST endpoint to query all post meta, even the meta values that aren't exposed to the Block Editor. This MUST be locked down.
- The sidebar should use [`usePostMeta`](https://github.com/alleyinteractive/alley-scripts/tree/main/packages/block-editor-tools/src/hooks/use-post-meta) to get a value that has been updated by the Block Editor, and fail back to the value from the REST endpoint.

Bonus points:
- Add a button to view meta values in a modal in addition to the sidebar, so that large meta values can be viewed easier
- Add JSON Formatter - if a value is JSON, or expressed as JSON because it's a serialized array/object in PHP, pretty-print the JSON to make it easier to read

**Comment by kevinfodness** [comment]
I'd recommend splitting the implementation steps into separate issues so that we can ship something initially that swaps the metabox for a PluginSidebar that shows meta visible to/managed by the editor, then iterates on that solution by adding the hidden meta, adding an open in a modal button, and adding the JSON prettifier.

**Comment by srtfisher** [comment]
This plugin does show information other than post meta including taxonomy terms, user/term meta, etc. It also displays on non-Gutenberg posts such as attachments, terms, and users. I would want to make sure that non-block editor pages still work properly. Perhaps there is a single solution that can support both but it seems that we'll need to support two paths: block editor slotfill for the block editor and the current meta boxes for non-block editor pages.

**Comment by mogmarsh** [comment]
@srtfisher Good call. I agree. I definitely don't want to lose any existing functionality either.
